scripts/extensions/auto-tagging-widget/src/auto-tagging.tsx

```diff
@@ -395,26 +395,91 @@ export function getAutoTaggingComponent(superdesk: ISuperdesk, label: string) {
 
                                 const entities = items.filter((tag) => isEntity(tag));
                                 const entitiesGrouped = entities.groupBy((tag) => tag?.group.value);
-                                const entitiesGroupedAndSorted = entitiesGrouped.sortBy(
-                                    (_, key) => key!.toString().toLocaleLowerCase(),
-                                    (a, b) => a.localeCompare(b),
-                                );
+
+                                const entitiesGroupedAndSortedByConfig = entitiesGrouped
+                                    .filter((_, key) => hasConfig(key, this.iMatricsFields.entities))
+                                    .sortBy((_, key) => this.iMatricsFields.entities[key].order,
+                                        (a, b) => a - b);
+
+                                const entitiesGroupedAndSortedNotInConfig = entitiesGrouped
+                                    .filter((_, key) => !hasConfig(key, this.iMatricsFields.entities))
+                                    .sortBy((_, key) => key!.toString().toLocaleLowerCase(),
+                                        (a, b) => a.localeCompare(b));
+
+                                const entitiesGroupedAndSorted = entitiesGroupedAndSortedByConfig
+                                    .concat(entitiesGroupedAndSortedNotInConfig);
 
                                 const others = items.filter((tag) => isEntity(tag) === false);
                                 const othersGrouped = others.groupBy((tag) => tag.group.value);
 
-                                const othersGroupedAndSortedByConfig = othersGrouped
-                                    .filter((_, key) => hasConfig(key, this.iMatricsFields))
-                                    .sortBy((_, key) => this.iMatricsFields[key].order,
+                                let allGrouped = OrderedMap<string, any>();
+
+                                othersGrouped.map((tags, groupId) =>
```

Comment:
return value of `map` isn't used

Reply:
what is not used? both `tags` and `groupId` is used it code below

Reply:
this is iteration for `othersGrouped` to be stored in `allGrouped` OrderedMap

Reply:
> what is not used?

The return value. `map` is a function that transforms an array/iterable, using the provided function and returns the new array. You never save that new array to a variable. If you only want to iterate the array/iterable, and not transform it, you should be using `forEach` instead.